From: Bart Van Assche <bvanassche@acm.org>
Date: Mon, 20 Dec 2021 09:23:48 -0800
Subject: net-snmp-create-v3-user: Fix the snmpd.conf path

@datadir@ is expanded in ${datarootdir} so datarootdir must be set before
@datadir@ is used. See also
https://github.com/net-snmp/net-snmp/pull/394 .

Fixes: d784eb53f38a ("NEWS: misc: separate user management from net-snmp-config into new net-snmp-create-v3-user script")

Origin: upstream, https://github.com/net-snmp/net-snmp/commit/72dee028836d233979e4563ce3c3d5e303e0e042
Bug-Ubuntu: https://bugs.launchpad.net/debian/+source/net-snmp/+bug/1969623
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=997895
---
 net-snmp-create-v3-user.in | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/net-snmp-create-v3-user.in b/net-snmp-create-v3-user.in
index b0c71d9..19895a1 100644
--- a/net-snmp-create-v3-user.in
+++ b/net-snmp-create-v3-user.in
@@ -134,8 +134,10 @@ if test ! -d "$outfile"; then
     touch "$outfile"
 fi
 echo "$line" >> "$outfile"
-# Avoid that configure complains that this script ignores @datarootdir@
-echo "@datarootdir@" >/dev/null
+prefix=@prefix@
+datarootdir=@datarootdir@
+# To suppress shellcheck complaints about $prefix and $datarootdir.
+: "$prefix" "$datarootdir"
 outfile="@datadir@/snmp/snmpd.conf"
 line="$token $user"
 echo "adding the following line to $outfile:"
